integrations/crowdnode/src/main/java/org/dash/wallet/integrations/crowdnode/transactions/FullCrowdNodeSignUpTxSet.kt

```diff
@@ -32,20 +32,54 @@ open class FullCrowdNodeSignUpTxSet(
         signUpFilter,
         CrowdNodeAcceptTermsResponse(networkParams),
         CrowdNodeAcceptTermsTx(networkParams),
-        CrowdNodeWelcomeToApiResponse(networkParams)
+        CrowdNodeWelcomeToApiResponse(networkParams),
+        PossibleAcceptTermsResponse(bag, null),
+        PossibleWelcomeResponse(bag, null)
     )
 
     private val matchedFilters = mutableListOf<TransactionFilter>()
     override val transactions = sortedSetOf(TransactionComparator())
 
     open val hasAcceptTermsResponse: Boolean
-        get() = matchedFilters.any { it is CrowdNodeAcceptTermsResponse }
+        get() = matchedFilters.any {
+            it is CrowdNodeAcceptTermsResponse ||
+                    (it is PossibleAcceptTermsResponse && didSignUpFromAddress(it.toAddress))
+        }
 
     open val hasWelcomeToApiResponse: Boolean
-        get() = matchedFilters.any { it is CrowdNodeWelcomeToApiResponse }
+        get() = matchedFilters.any {
+            it is CrowdNodeWelcomeToApiResponse ||
+                    (it is PossibleWelcomeResponse && didSignUpFromAddress(it.toAddress))
+        }
 
     open val accountAddress: Address?
-        get() = (matchedFilters.firstOrNull { it is CrowdNodeSignUpTx } as? CrowdNodeSignUpTx)?.fromAddresses?.first()
+        get() {
+            matchedFilters.firstOrNull { it is CrowdNodeWelcomeToApiResponse }?.let {
```

Comment:
Added a better account detection logic that priorities responses with full info about connected transactions.